engines/registry/lib/registry/engine.rb

```diff
@@ -15,7 +15,7 @@ class Engine < ::Rails::Engine
         before_action :handle_auth_cache, only: %w[index]
```

Comment:
I think it would be more readable like this: 

```suggestion
        # only run instance verification and cache refresh in PAYG mode
        before_action :refresh_auth_cache, only: %w[index], if: -> { request.headers['X-Instance-Data'] }

        def refresh_auth_cache
          unless ZypperAuth.verify_instance(request, logger, @system)
            render(xml: { error: 'Instance verification failed' }, status: :forbidden)
          end
        end
```

Reply:
That's a really elegant fix @tomschr .